doc/design/GeneratedClientDesign.md

```diff
@@ -0,0 +1,133 @@
+# Generated Client Design
+
+## Introduction
+
+This document tracks design rules and decisions for .NET clients generated by the DPG generator.  It may later be extended to include other generator designs.
+
+<!--
+## Clients
+
+TBD
+
+## Client Protocol Operations
+
+TBD
+
+## Client Convenience Operations
+-->
+
+## Model Generation
+
+Models are defined as partial classes with one or more constructors, properties, and methods that read from/write to wire formats.
+
+### Model Namespace
+
+Generated models will be defined in a `.Models` namespace.
+
+### Type mappings from Cadl to .NET
+
+The following table shows a mapping from Cadl built-in types to the corresponding types used in .NET models.
+
+Cadl Type | .NET Type | OpenAPI Type | GitHub Issue | Notes
+------------------- | -------- | -- | -- | -------------
+string | string | string | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) | [PrimitivePropertyModel.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-model-primitive-properties/sdk/template/Azure.Template/src/Generated/Models/PrimitivePropertyModel.cs) [PrimitivePropertyModel.Serialization.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-model-primitive-properties/sdk/template/Azure.Template/src/Generated/Models/PrimitivePropertyModel.Serialization.cs)
+bytes | bytes[] | type: string, format: byte | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
+int32  | int | type: integer, format: int32 | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
+int64  | long | type: integer, format: int64 | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
+safeint  | long | n/a | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
+float32  | float | type: number, format: float | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
```

Comment:
Only negative of using double is if these are stored in an array.

If float32 is on a class, make it a double.

Decision: keep as a float.